Refactor: 서비스 로직, Review의 별점

Minjeong63 opened the pull request:
### 변경사항
- AuctionService, TicketService에서 필요없는 @Transactional 제거, 내부 메서드 접근 제한자 public으로 변경
- reviewRating 관련 코드 전부 타입을 double  => Integer로 변경
- 식사권 취소할 때 수수료 계산하는 부분에 빠진 조건 추가
- swagger 테스트 코드 삭제

### 고민되는 부분
- 현재 리뷰 작성 API에서 reviewRating 타입을 int로 지정했으나, double 타입이 들어오더라도 소수점 자리가 버려지고 저장이 됩니다. 따로 처리해줘야 할까요? 아님 프론트에서 정수로 보낼 수 있게만 처리하면 될까요?

Minjeong63 committed:
Refactor: 서비스 로직, Review의 별점
Review : reviewRating 타입을 double => int로 변경
AuctionService : 필요없는 @Transactional 제거, 내부 메서드 접근 제한자 public으로 변경
TicketService : 필요없는 @Transactional 제거, 내부 메서드 접근 제한자 public으로 변경, 수수료 계산하는 로직에 조건 추가
TicketServiceTest : 수수료 계산 테스트

Minjeong63 committed:
Merge branch 'develop' into feature/review

Minjeong63 committed:
Refactor: ErrorCode

dnjs2721 commented:
확인했습니다.
프론트에서 정수로 보내는걸로 해야할겁니다.

Minjeong63 commented:
넵 알겠습니다!

dnjs2721 commented:
리뷰 작성시 리뷰 대상자 레이팅 수정은 된건가요? 해당 로직이 안보이네요.

Minjeong63 commented:
리뷰 대상자 레이팅 수정은 어떤 걸 말씀하시는 거죠?

Minjeong63 commented:
사용자 별점 계산을 말씀하시는 건가요?

dnjs2721 commented:
리뷰 작성시 사용자 테이블의 rating을 수정하는걸로 알고있습니다.
해당 로직이 안보여서요.
추가로 리뷰 세이브로직에서 ticket의 UUID가 아닌 id를 쓴 이유가 있을까요?

dnjs2721 commented:
> 사용자 별점 계산을 말씀하시는 건가요?

넵

Minjeong63 commented:
PathVariable로 넘어오는 값이 아니고 body로 넘어오는 데이터라서 id로 했습니다.
id로 한 특별한 이유가 있는 건 아닌데 UUID로 할 필요가 없는 것 같아서 id로 했습니다.

Minjeong63 commented:
사용자 별점 로직은 추가하겠습니다

Minjeong63 commented:
사용자의 별점 변경도 락이 걸리는 게 맞겠죠?

dnjs2721 commented:
프론트는 식사권 세부 정보를 받아올때 id 혹은 UUID를 전달 받나요?
해당 부분이 있어야 다른 api에서 id, UUID 로 데이터를 보낼수있습니다.

dnjs2721 commented:
> 사용자의 별점 변경도 락이 걸리는 게 맞겠죠?

별점 변경은 동시성 문제가 상대적으로 적어보입니다.
필수는 아니라고 봅니다.

Minjeong63 commented:
아 넵 리뷰는 제 생각에도 필수는 아닌 것 같네요 따로 락을 걸진 않겠습니다

dnjs2721 commented:
> 프론트는 식사권 세부 정보를 받아올때 id 혹은 UUID를 전달 받나요? 해당 부분이 있어야 다른 api에서 id, UUID 로 데이터를 보낼수있습니다.

이 부분은 티켓 세부 정보에서 ticketUUID를 타고 들어와서 괜찮을거 같네요.
다른 티켓 관련 api에서 id를 사용하기 위해선 ticketid를 세부정보에 포함시켜야 될 거 같습니다.

Minjeong63 commented:
PathVariable에서는 ticketUUID를 사용하니 id를 보내는 부분을 다 ticketUUID로 변경하겠습니다. 그게 프론트에서 데이터 보낼 때도 일관성 있어서 좋을 것 같네요

dnjs2721 commented:
넵 감사합니다.
추가 브랜치 작업하시나요?

Minjeong63 commented:
앗 저도 모르게 review 브랜치에서 하고 있었습니다..

dnjs2721 commented:
그대로 작업하셔도 될거같습니다.
해당 브랜치에 푸시하시죠

Minjeong63 commented:
테스트까지 하고 내일 아침에 푸시해도 될까요?

dnjs2721 commented:
넵 괜찮습니다~

Minjeong63 committed:
Refactor: 리뷰 작성 로직에 사용자 별점 update 로직 추가, TicketService의 reqDto에 있던 ticketId => ticketUUID로 변경
req에 있던 ticketId => ticketUUID로 변경
User : 리뷰의 receiver 양방향 매핑, updateRating 메서드 추가
ErrorCode, GlobalExceptionHandler에 있던 ticketId => ticketUUID

dnjs2721 commented:
확인했습니다

justashow commented:
확인했습니다
